Pages/PlayAlong.elm

```diff
@@ -29,17 +39,42 @@ bars =
        , rhythmDots
        ] |> group
 
+chordNames : ChordProgression -> Form
+chordNames =
+    let f mm nq = uncurry nameOfRelativeChord nq
+               |> Text.fromString
+               |> text
+               |> moveToTime mm 8
+               |> moveY (barHeight / 8)
+    in group << List.map2 f barsList << Array.toList
+
 rhythmDots : Form
 rhythmDots =
     let dot mm sq = circle 3
                  |> outlined (solid black)
-                 |> moveToTime mm (sq - 1)
+                 |> moveToTime mm (sq + 2)
+                 |> moveY (toFloat <| -barHeight // 4)
     in group
           << List.concat
-          << flip   List.map          [0 .. barsNumX * barsNumY - 1]
-          <| \mm -> List.map (dot mm) [4, 8, 12, 16]
-
+          << flip   List.map          barsList
+          <| \mm -> List.map (dot mm) [0, 4, 8, 12]
 
+chordToPlay : Note -> ChordProgression -> Int -> Form
+chordToPlay tonic prog mm =
+    let idx = mm % (barsNumX * barsNumY)
+        (num, qual) = case Array.get idx prog of
+            Just x -> x
+            Nothing -> (0, Maj)
+        -- TODO(sandy): I'm not convinced this is the
```

Comment:
it turns out it is*, unless we want to go down a big spooky rabbit hole. 

* so actually what happens is if you start in a minor key, you need to transform eg. a minor third into a major key because the major key is minor relative to a minor, and.... well let's not get into it